content/en/device-provisioning-service/client-library.md

```diff
@@ -50,49 +51,75 @@ void plgd_dps_shutdown(void);
 plgd_dps_context_t *plgd_dps_get_context(size_t device);
 
 /// @brief Set DPS manager callbacks.
-void plgd_dps_manager_init_callbacks(plgd_dps_context_t *ctx, plgd_dps_on_status_change_cb_t on_change_cb,
-                                     void *on_change_data, oc_cloud_cb_t on_cloud_change_cb,
-                                     void *on_cloud_change_data);
+void plgd_dps_set_manager_callbacks(plgd_dps_context_t *ctx, plgd_dps_on_status_change_cb_t on_change_cb,
+  void *on_change_data, oc_cloud_cb_t on_cloud_change_cb, void *on_cloud_change_data);
 
 /**
  * @brief Start DPS manager to provision device by given server.
  *
- * Setup context, handlers, dps server and start dps manager.
+ * Setup context, handlers, dps endpoint and start dps manager.
  *
- * @param ctx device context (cannot be NULL)
+ * @param ctx dps context (cannot be NULL)
  * @param endpoint endpoint of the provisioning server (cannot be NULL)
  * @return 0 on success
  * @return -1 on failure
  */
-int plgd_dps_manager_start(plgd_dps_context_t *ctx, const char *endpoint);
+int plgd_dps_manager_start(plgd_dps_context_t *ctx);
```

Comment:
where is documented retry mechanism? Or is it still under development?

Reply:
Documentation added.

Reply:
Wasn't there a method to callback to trigger retry on your own?

Reply:
You can trigger re-provisioning by using the `rst` property of the `/plgd/dps` resource; 

In the `dps_cloud_server` device you have also the option to send `SIGHUP` signal, which force re-provisioning.

Or are you thinking of something else?